### ISSUE TITLE
Invalid Scala code generated if package names contain Scala reserved words

### ISSUE THREAD

**Issue by blissd** [scalapb/ScalaPB#499]
I have a protobuf that defines a package something like this:

```protobuf
package foo_override_bar_v1;
```
Which generates the following Scala code:
```scala
package foo.override.bar.v1
```
However, that package statement is invalid, because `override` is a Scala key word.

Could the code generator escape Scala keywords? For example:

```scala
package foo.`override`.bar.v1
```

**Comment by blissd** [comment]
My `project/scalapb.sbt` is:

```scala
addSbtPlugin("com.thesamet" % "sbt-protoc" % "0.99.18")

libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.8.0"
```

**Comment by thesamet** [comment]
I can't reproduce this issue. If the package name is as stated in the original message (`foo_override_bar_v1`), then the generated code becomes:

```
package foo_override_bar_v1
```

If the package name contains dots `foo.override.bar.v1`, then ScalaPB emtis:

```
package foo.`override`.bar.v1
```

which is the correct output. The examples above assume `flat_package` is true, but work also when it is set to false. Can you double check and maybe provide a complete example?

**Comment by blissd** [comment]
Steps to reproduce:

Create a file at `src/main/protobufs/mode.override.v1.proto`.

Contents of file:

```protobuf
syntax = "proto3";

package Mode_override_v1;

option java_package = "com.example.protobuf";

enum Mode_override_v1 {
    NONE = 0;
    TRANSIENT = 1;
    STANDBY = 2;
    EMERGENCY = 3;
}
```

Two Scala files are generated. In `Mode_override_v1` the package statement is invalid because of the `override` in the package name:

```scala

package com.example.protobuf.mode.override.v1
 ..
```

The same is true for the second Scala file `ModeOverrideV1Proto1`.

I think my original analysis was incorrect. The problem appears to be the `override` in the file name, not the `override` in the package name.

**Comment by blissd** [comment]
I think this line is the problem: https://github.com/scalapb/ScalaPB/blob/d03e74e0f36c61fc899588382b46b65ef1578b33/compiler-plugin/src/main/scala/scalapb/compiler/DescriptorImplicits.scala#L694

**Comment by blissd** [comment]
Hey... just noticed that you also wrote the Scala KD-tree. We use that too, thanks!

**Comment by blissd** [comment]
It would be great if that PR could be merged and a new release cut.

**Comment by thesamet** [comment]
Released 0.8.1
